app/controllers/casino/application_controller.rb

```diff
@@ -22,8 +22,18 @@ def processor(processor_name, listener_name = nil)
     @processor = CASino.const_get(:"#{processor_name}Processor").new(listener)
   end
 
+  def available_locale
+    ['de','en']
+  end
+
   def set_locale
-    I18n.locale = extract_locale_from_accept_language_header || I18n.default_locale
+    locale = if params[:service] and params[:service].match(/.*[?&]locale=([^&]+)(&|$)/)
+               result = params[:service].match(/.*[?&]locale=([^&]+)(&|$)/)
```

Comment:
I believe that it could be optimized slightly: 1) the initial `.*` is not necessary, because the regexp is not anchored at the beginning, and we do not later need the whole matched string. 2) there is no need to call the `match` twice, as the first call (in the `if`) could store the result in a local variable.

_(This also can be done any time later)_